accepted/2021/statics-in-interfaces/README.md

```diff
@@ -6,8 +6,1267 @@
+
+        static abstract TResult operator *(TSelf lhs, TOther rhs);
+
+        static abstract TResult checked operator *(TSelf lhs, TOther rhs);
+    }
+
+    public interface IDivisable<TSelf, TOther, TResult>
+        where TSelf : IDivisable<TSelf, TOther, TResult>
+    {
+        static abstract TResult operator /(TSelf lhs, TOther rhs);
+
+        static abstract TResult checked operator /(TSelf lhs, TOther rhs);
+    }
+
+    public interface IRemainder<TSelf, TOther, TResult>
+        where TSelf : IRemainder<TSelf, TOther, TResult>
+    {
+        // The ECMA name is op_Modulus and so the behavior isn't actually "remainder" with regards to negative values
+
+        // Likewise the name IRemainder doesn't fit with the other names, so a better one is needed
+
+        static abstract TResult operator %(TSelf lhs, TOther rhs);
+
+        static abstract TResult checked operator %(TSelf lhs, TOther rhs);
+    }
+
+    public interface INegatable<TSelf, TResult>
+        where TSelf : INegatable<TSelf, TResult>
+    {
+        // Should unary plus be on its own type?
+
+        static abstract TResult operator +(TSelf value);
+
+        static abstract TResult checked operator +(TSelf value);
+
+        static abstract TResult operator -(TSelf value);
+
+        static abstract TResult checked operator -(TSelf value);
+    }
+
+    public interface IFixedWidth<TSelf>
+        where TSelf : IFixedWidth<TSelf>
+    {
+        // MinValue and MaxValue are more general purpose than just "numbers", so they live on this common base type
+        // Fixed-Width isn't the best name however, so we should think of alternatives
+
+        static abstract TSelf MinValue { get; }
+
+        static abstract TSelf MaxValue { get; }
+    }
+}
+```
+
+### Numeric Interfaces
+
+The numeric interfaces build upon the base interfaces by defining the core abstraction that most users are expected to interact with.
+
+```csharp
+namespace System
+{
+    public interface INumber<TSelf>
+        : IAddable<TSelf, TSelf>,
+          IComparableOperators<TSelf, TSelf>,   // implies IEquatableOperators<TSelf, TSelf>
+          IDecrementable<TSelf>,
+          IDivisable<TSelf, TSelf>,
+          IIncrementable<TSelf>,
+          IMultipliable<TSelf, TSelf>,
+          IRemainder<TSelf, TSelf>,
+          ISpanFormattable,                     // implies IFormattable
+          ISpanParseable<TSelf>,                // implies IParseable<TSelf>
+          ISubtractable<TSelf, TSelf>
+        where TSelf : INumber<TSelf>
+    {
+        // For the Create methods, there is some concern over users implementing them. It is not necessarily trivial to take an arbitrary TOther
+        // and convert it into an arbitrary TSelf. If you know the type of TOther, you may be able to optimize in a few ways. Otherwise, you
+        // may have to fail and throw in the worst case.
```

Comment:
I think it entirely depends on the types you are wanting to support and consume.

I'd expect it can succeed most of the time, but may not be fast for uncommon types. You would only need to fail if a user provided some type for which there is no way to get the necessary value to use in the conversion.